aws-samples/emr-tls-provider-sercets-manager-example
# feat/upgrade

*Description of changes:*

Major changes:

- Introduce dedicated components for certificate, EC2, secrets, and TLS management
- Add robust retry mechanism with RetryHandler
- Implement proper exception handling with specific exception types
- Add development container support and Makefile for build automation
- Configure logging with logback
- 
This refactoring improves code organization, maintainability, and reliability by separating concerns into specialized components and adding proper error handling throughout the codebase.

- Add integration tests to verify TLS setup
- Generate PKCS#8 certificates for local testing
- Create and verify keystore/truststore
- Update .gitignore to exclude test certificates
- Add make targets for certificate generation and testing
- Implement end-to-end TLS verification

This allows testing TLS configuration locally before EMR deployment.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
